generic.go

```diff
@@ -57,17 +67,50 @@ func NewGeneric(config *Config, logger *zap.Logger, queue workqueue.RateLimiting
 			return nil, errors.Errorf("controller for GVK %s should have registered an informer for that GVK", descr.Gvk)
 		}
 		inf.AddEventHandler(&GenericHandler{
-			Logger:       logger,
+			Logger:       gvkLogger,
 			WorkQueue:    queueGvk,
 			ZapNameField: descr.ZapNameField,
 		})
 		controllers[descr.Gvk] = iface
+
+		// Extra controller data
+		groupKind := descr.Gvk.GroupKind()
+		replacer := strings.NewReplacer(".", "_", "-", "_", "/", "_")
```

Comment:
Should be outside of the loop - need only one.

Reply:
Good point